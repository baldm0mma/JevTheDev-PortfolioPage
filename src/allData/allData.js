export const projectContent = [
  {
    id: 1,
    title: "My Eighties Computer (NPM Package)",
    description:
      "This is the first NPM Package I've ever written! I love contributing to free, opensource code, and I'm really excited for this to be my first home-baked contribution! There are four components in total, a styled h1, a styled paragraph text, and two styled and controlled forms. The h1 and paragraph texts take a single prop, which is the text that they will display; and the two form components take two props each; one is the text title for the form, and the second is a callback function that returns the current controlled state of the form inputs. It was a blast to make, and I hope someone gets some use out of it!",
    gitHub: "https://github.com/baldm0mma/my-eighties-computer",
    live: "https://baldm0mma.github.io/my-eighties-computer/",
    image: "./assets/My80sComputer.png",
  },
  {
    id: 2,
    title: "Home Safe",
    description:
      "Rome wasn't built in a day; but this app was! A finalist for the 2019 Tech to Protect Hackathon, Home Safe gives homeowners an easy way to upload their pertinent safety information to be stored and displayed in a useful manner. It encourages users to complete a safety checklist, upload floor plans, and create a pre-incident plan to be printed and saved for display in the home. All of these features culminate to help homeowners get insurance breaks and offer first responders an amazing set of data that will help them do their job.",
    gitHub: "https://github.com/baldm0mma/home-safe",
    live: "https://ttp-home-safe.herokuapp.com/",
    image: "./assets/HomeSafe.gif",
  },
  {
    id: 3,
    title: "Ride with Me",
    description:
      "Ride With Me is a mobile application written in React Native that serves as a community board for motorcyclists to connect with other riders. Upon creation of an account, users can log in, add friends, and join rides based on criteria that they so choose, such as location, distance, and type of ride. Check out our front end code on GitHub (featuring React Native, Redux, Apollo, and XCode), and our back end GraphQL database on GitLab.",
    gitHub: "https://github.com/baldm0mma/ride_with_me",
    gitLab: "https://gitlab.com/ryanmillergm/motorcycle_ride",
    image: "./assets/RideWithMe.gif",
  },
  {
    id: 4,
    title: "Richard Mortimer",
    description:
      "This project is Rick and Morty fan app. With the announcement of Rick and Morty Season 4 coming this winter, there's a lot of excitement around it's hotly anticipated release. This app is designed to be a endorsement piece; an attempt to keep up the excitement for the upcoming season. There's a bit of gamification, a bit of shameless plugging of Adult Swim, and a LOT of 80's computer styling.",
    gitHub: "https://github.com/baldm0mma/richard_mortimer",
    live: "https://rich-mort.herokuapp.com/",
    image: "./assets/RichardMortimer.gif",
  },
  {
    id: 5,
    title: "Kouleur-FE",
    description:
      "Kouleur-FE is the React front-end of a full-stack color palette picker application. We used Semantic UI elements to style and design an elegant, thoughtful, and intuitive UI/UX experience. This front end app pairs with a PostgreSQL/Node.js/Express.js RESTful API which we also built our ourselves that utilizes semantic HTTP methods, intuitive url pathway naming conventions, and a one-to-many database of our own design. You can find a link to the back end on our front end GitHub page.",
    gitHub: "https://github.com/baldm0mma/kouleur-fe",
    live: "https://kouleur.herokuapp.com/",
    image: "./assets/Kouleur.gif",
  },
  {
    id: 6,
    title: "NASA POTD in Vue Challange",
    description:
      "This project was takehome code challenge. The prompt goes as follows: \"Build a single-page application in Vue.js that queries the NASA Astronomy Picture of the Day API and displays the day's image. The application should include a way to view each of the current month's pictures (it's okay if the month only has 1 photo so far, for example if it's the first of the month), and a way to return to the current day's picture. Be creative with how you present the collections - think about the experience of the viewer with what they want to view and learn.\"",
    gitHub: "https://github.com/baldm0mma/NASA_APOD_vue",
    gitLab: "",
    live: "https://nasa-apod-final.herokuapp.com/",
    image: "./assets/VueNasa.gif",
  },
];

export const aboutContent = {
  id: 8,
  name: "Jev Forsberg",
  title: "Software Engineer || Educator || Civic Innovator/Advocate",
  description:
    "I was a civic advocate, nonprofit manager, and pastor for almost a decade, and I absolutely loved the mentorship, community building, teaching, service, team-leading, public speaking, and civil-rights work I did. In 2018, I found another interest that I've become incredibly passionate about: technology; and specifically software engineering. No matter your background, context, region, language, race, religion, ability, gender expression, or any other category we use to label one another, there's a place for you in the tech community, and that's one of the many things I love about it!",
  gitHub: "https://github.com/baldm0mma",
  linkedIn: "https://www.linkedin.com/in/jev-forsberg/",
  resume:
    "https://drive.google.com/file/d/1BrA5L4kzf9m52y_0DiLJNglrA8EmA0ud/view?usp=sharing",
  email: "mailto:jev.forsberg@gmail.com",
  image: "./assets/Profile-Gif.gif",
  skills: [
    "JavaScript/TypeScript",
    "React.js/Vue.js/React Native",
    "Node.js/Express.js",
    "RESTful/CRUD APIs/GraphQL",
    "PostgreSQL/Knex.js/SQL",
    "Elixir/Erlang/OTP",
    "ElasticSearch",
    "Kubernetes",
  ],
};

export const contractContent = [
  {
    title: "SchoolSAFE Communications",
    dates: "September 2018 - December 2019",
    description:
      "Reengineered an obsolete code base for Colorado's public school's to have direct communication with emergency services in the event of an emergency. Our team rebuilt a MySQL database with Node.js and PostgreSQL, redesigned a Flash and PHP front end with React.js and Redux, and established communication between front end systems and IoT devices. Featured Bcrypt, JWT, and Secure Cookies for authentication and sessioning.",
    image: "./assets/SchoolSAFE.gif",
  },
  {
    title: "Bryan Barton Construction",
    dates: "September 2019 - January 2020",
    description:
      "Designed, built, and implemented a dynamic, full-stack, and multi-interface web application that connects homeowners and property managers with construction and home repair needs with builders and contractors. Homeowners/property managers, contractors, and administrators all have separate logins and permissions; Homeowners/property managers can enter a job, then contractors/builders can enter a bid, finally a bid is chosen, and the job is archived.",
    image: "./assets/under-construction.png",
  },
];
