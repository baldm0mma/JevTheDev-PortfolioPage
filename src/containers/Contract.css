.contract-container {
  background-attachment: fixed;
  background-image: linear-gradient(
      rgba(255, 255, 255, 0.5),
      rgba(255, 255, 255, 0.5)
    ),
    url("../backgrounds/background-light-grey.jpg");
  background-position: center;
  background-repeat: no-repeat;
  background-size: 2000px;
  border-radius: 30px;
  display: flex;
  margin-bottom: 120px;
  text-shadow: white 8px 8px 25px;
  width: auto;
}

.contract-info-container {
  display: flex;
  flex-direction: column;
  margin: 20px;
  width: 350px;
}

.contract-image-container {
  align-items: center;
  display: flex;
  justify-content: center;
}

.contract-image {
  max-height: 400px;
  max-width: 400px;
  padding: 30px;
}

@media only screen and (max-width: 1000px) {
  .contract-container {
    flex-direction: column;
  }

  .contract-image {
    align-self: center;
    max-width: 475px;
  }

  .contract-info-container {
    align-self: center;
    width: 475px;
  }
}

@media only screen and (max-width: 600px) {
  .contract-container {
    width: 450px;
  }

  .contract-image {
    max-width: 350px;
  }

  .contract-info-container {
    align-self: center;
    width: 350px;
  }
}

@media only screen and (max-width: 450px) {
  .contract-container {
    width: 350px;
  }

  .contract-image {
    max-width: 300px;
  }

  .contract-info-container {
    align-self: center;
    width: 300px;
  }
}
