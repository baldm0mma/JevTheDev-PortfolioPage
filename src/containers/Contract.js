import React from 'react';
import { contractContent } from '../projectContent/projectContent';

const { title, dates, description, image } = contractContent;

export const Contract = () => {
  return (
    <article className='contract-container'>
      <section>
        <h3 className='contract-title'>{title}</h3>
        <h4 className='contract-dates'>Contract Dates: {dates}</h4>
        <p className='contract-description'>{description}</p>
      </section>
      <img
        className='contract-image'
        src={image}
        alt={`photo of ${title} contract project`}
      />
    </article>
  );
};
