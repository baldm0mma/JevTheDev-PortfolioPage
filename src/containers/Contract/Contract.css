.contract-container {
  background-attachment: fixed;
  background-image: linear-gradient(
      rgba(255, 255, 255, 0.5),
      rgba(255, 255, 255, 0.5)
    ),
    url("../../backgrounds/background-light-grey.jpg");
  background-position: center;
  background-repeat: no-repeat;
  background-size: 2000px;
  border-radius: 30px;
  display: flex;
  align-items: stretch;
  margin-bottom: 120px;
  text-shadow: white 8px 8px 25px;
  width: auto;
  min-height: 200px;
}

.contract-info-container {
  display: flex;
  flex-direction: column;
  margin: 20px;
  flex: 2;
  min-width: 300px;
  max-width: 600px;
}

.contract-image-container {
  align-items: center;
  display: flex;
  justify-content: center;
  flex: 1;
  min-width: 200px;
}

.contract-image {
  max-height: 300px;
  max-width: 100%;
  height: auto;
  width: auto;
  padding: 20px;
  object-fit: contain;
}

@media only screen and (max-width: 1000px) {
  .contract-container {
    flex-direction: column;
  }

  .contract-image {
    align-self: center;
    max-width: 400px;
    max-height: 300px;
  }

  .contract-info-container {
    align-self: center;
    max-width: 475px;
    min-width: 300px;
  }

  .contract-image-container {
    flex: none;
  }
}

@media only screen and (max-width: 600px) {
  .contract-container {
    max-width: 450px;
    margin: 0 auto 120px auto;
  }

  .contract-image {
    max-width: 300px;
    max-height: 250px;
  }

  .contract-info-container {
    align-self: center;
    max-width: 350px;
    min-width: 280px;
  }
}

@media only screen and (max-width: 450px) {
  .contract-container {
    max-width: 350px;
    margin: 0 auto 120px auto;
  }

  .contract-image {
    max-width: 250px;
    max-height: 200px;
  }

  .contract-info-container {
    align-self: center;
    max-width: 300px;
    min-width: 250px;
  }
}
