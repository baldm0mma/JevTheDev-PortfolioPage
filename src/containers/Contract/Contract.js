import React from "react";
import "./Contract.css";

export const Contract = ({ content }) => {
  const { title, dates, description, image } = content;

  return (
    <article className="contract-container">
      <section className="contract-info-container">
        <h3 className="contract-title" data-testid="contract-title">
          {title}
        </h3>
        <h4 className="contract-dates" data-testid="contract-dates">
          Contract Dates: {dates}
        </h4>
        <p className="contract-description" data-testid="contract-description">
          {description}
        </p>
      </section>
      <article className="contract-image-container">
        <img
          className="contract-image"
          src={image}
          alt={`${title} contract project`}
        />
      </article>
    </article>
  );
};
