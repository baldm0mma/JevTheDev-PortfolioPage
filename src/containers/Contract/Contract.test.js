import React from "react";
import { shallow } from "enzyme";
import { Contract } from "./Contract";
import { contractContent } from "../../allData/allData.js";

describe("Contract Component", () => {
  let wrapper;
  const { title, dates, description, image } = contractContent[0];

  beforeEach(() => {
    wrapper = shallow(<Contract content={contractContent[0]} />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should properly render the dynamic element texts", () => {
    expect(wrapper.find("[data-testid='contract-title']").text()).toBe(title);
    expect(wrapper.find("[data-testid='contract-dates']").text()).toBe(
      `Contract Dates: ${dates}`
    );
    expect(wrapper.find("[data-testid='contract-description']").text()).toBe(
      description
    );
  });

  it("should properly assign a dynamic image source", () => {
    expect(wrapper.find(`[src="${image}"]`).exists()).toBe(true);
  });
});
