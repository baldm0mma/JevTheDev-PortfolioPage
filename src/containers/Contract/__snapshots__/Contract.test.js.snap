// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract Component should match the snapshot 1`] = `
<article
  className="contract-container"
>
  <section
    className="contract-info-container"
  >
    <h3
      className="contract-title"
      data-testid="contract-title"
    >
      SchoolSAFE Communications
    </h3>
    <h4
      className="contract-dates"
      data-testid="contract-dates"
    >
      Contract Dates: 
      September 2019 - Present
    </h4>
    <p
      className="contract-description"
      data-testid="contract-description"
    >
      Reengineered an obsolete code base for Colorado's public school's to have direct communication with emergency services in the event of an emergency. Our team rebuilt a MySQL database with Node.js and PostgreSQL, redesigned a Flash and PHP front end with React.js and Redux, and established communication between front end systems and IoT devices. Featured Bcrypt, JWT, and Secure Cookies for authentication and sessioning.
    </p>
  </section>
  <article
    className="contract-image-container"
  >
    <img
      alt="SchoolSAFE Communications contract project"
      className="contract-image"
      src="./assets/SchoolSAFE.gif"
    />
  </article>
</article>
`;
